internal/dkg/store.go

```diff
@@ -24,42 +24,107 @@ type BoltStore struct {
+const dkgStagedFileName = "dkg.staged.toml"
 const BoltStoreOpenPerm = 0660
 const DirPerm = 0755
 
 var stagedStateBucket = []byte("dkg")
 var finishedStateBucket = []byte("dkg_finished")
 
-func NewDKGStore(baseFolder string, options *bolt.Options) (*BoltStore, error) {
-	err := os.MkdirAll(baseFolder, DirPerm)
+type FileStore struct {
+	baseFolder string
+	log        log.Logger
+}
+
+func NewDKGStore(baseFolder string) (*FileStore, error) {
+	err := os.MkdirAll(baseFolder, 0770)
+	if err != nil {
+		return nil, err
+	}
+	return &FileStore{
+		baseFolder: baseFolder,
+		log:        log.New(nil, log.DebugLevel, true),
+	}, nil
+}
+
+func getFromFilePath(path string) (*DBState, error) {
+	t := DBStateTOML{}
+	_, err := toml.DecodeFile(path, &t)
 	if err != nil {
 		return nil, err
 	}
-	dbPath := path.Join(baseFolder, BoltFileName)
-	db, err := bolt.Open(dbPath, BoltStoreOpenPerm, options)
+	state, err := t.FromTOML()
 	if err != nil {
 		return nil, err
 	}
+	return state, nil
+}
 
-	err = db.Update(func(tx *bolt.Tx) error {
-		_, err := tx.CreateBucketIfNotExists(stagedStateBucket)
-		if err != nil {
-			return err
-		}
+func (fs FileStore) GetCurrent(beaconID string) (*DBState, error) {
+	f, err := getFromFilePath(path.Join(fs.baseFolder, beaconID, dkgStagedFileName))
+	if errors.Is(err, os.ErrNotExist) {
+		fs.log.Debug("No DKG file found, returning new state")
+		return NewFreshState(beaconID), nil
+	}
+	return f, err
+}
+
+func (fs FileStore) GetFinished(beaconID string) (*DBState, error) {
+	f, err := getFromFilePath(path.Join(fs.baseFolder, beaconID, dkgFileName))
+	if errors.Is(err, os.ErrNotExist) {
+		return nil, nil
+	}
+	return f, err
+}
 
-		_, err = tx.CreateBucketIfNotExists(finishedStateBucket)
+func saveTOMLToFilePath(filepath string, state *DBState) error {
+	w, err := os.Create(filepath)
+	if err != nil {
 		return err
-	})
+	}
+	t := state.TOML()
+	err = toml.NewEncoder(w).Encode(&t)
 	if err != nil {
-		return nil, err
+		return err
 	}
+	return nil
+}
 
-	store := BoltStore{
-		db:  db,
-		log: log.New(nil, log.DebugLevel, true),
+// SaveCurrent stores a DKG packet for an ongoing DKG
+func (fs FileStore) SaveCurrent(beaconID string, state *DBState) error {
+	return saveTOMLToFilePath(path.Join(fs.baseFolder, beaconID, dkgStagedFileName), state)
+}
+
+// SaveFinished stores a completed, successful DKG and overwrites the current packet
+func (fs FileStore) SaveFinished(beaconID string, state *DBState) error {
+	return saveTOMLToFilePath(path.Join(fs.baseFolder, beaconID, dkgFileName), state)
+}
+
+func (fs FileStore) Close() error {
+	// Nothing to do for flat-file management
+	return nil
+}
+
+func (fs FileStore) MigrateFromGroupfile(beaconID string, groupFile *key.Group, share *key.Share) error {
+	beaconDir := path.Join(fs.baseFolder, beaconID)
+	fs.log.Debug(
+		"Attempting to migrate state for beaconID  %q, ensuring directory %s exists...",
+		beaconID, beaconDir)
+	_, err := os.Create(beaconDir)
```

Comment:
this ought to already be created by the keypair generation - it probably suggests operator error in beaconID if not (if they can even get an erroneous beaconID by here?)